src/VecSim/types/float16.h

```diff
@@ -0,0 +1,102 @@
+/*
+ *Copyright Redis Ltd. 2021 - present
+ *Licensed under your choice of the Redis Source Available License 2.0 (RSALv2) or
+ *the Server Side Public License v1 (SSPLv1).
+ */
+
+#pragma once
+
+#include <cstdint>
+#include <cstring>
+#include <algorithm>
+namespace vecsim_types {
+
+using float16 = uint16_t;
+
+inline float _interpret_as_float(uint32_t num) {
+    void *num_ptr = &num;
+    return *(float *)num_ptr;
+}
+
+inline int32_t _interpret_as_int(float num) {
+    void *num_ptr = &num;
+    return *(int32_t *)num_ptr;
+}
```

Comment:
```suggestion
#include <algorithm>
#include <bit>
namespace vecsim_types {

using float16 = uint16_t;

inline float _interpret_as_float(uint32_t num) {
    return std::bit_cast<float>(num);
}

inline int32_t _interpret_as_int(float num) {
    return std::bit_cast<int32_t>(num);
}
```

Reply:
nice!
so from which GCC version it is supported?

Reply:
this is c++20 and was added as of GCC 11

Reply:
so we cannot support in all OSs :(